### PR TITLE
prefer identifying accounts by Address instead of (signing) Key

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

**Comment by dgarcia360** [comment]
@Jaguar0625 There is an open comment in the proposal doc:

> Is it technically possible to change ``signerPublicKey`` fields to addresses? Will entities be less verifiable if this change is applied?

File: schemas/state/secret_lock.cats [hunk 1]
```diff
@@ -3,8 +3,8 @@ import "state/lock_info.cats"
 
 # binary layout for serialized lock transaction
 struct SecretLockInfo
-	# sender public key
-	senderPublicKey = Key
+	# owner address
+	ownerAddress = Address
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply schemas/state/secret_lock.cats:7]: we decided for consistency it should be owner to match naming in mosaic and namespace

File: schemas/state/multisig_entry.cats [hunk 2]
```diff
@@ -8,8 +8,8 @@ struct MultisigEntry
 	# minimum approval for removal
 	minRemoval = uint32
 
-	# account public key
-	accountPublicKey = Key
+	# account address
+	accountAddress = Address
 
 	# number of cosignatories
 	cosignatoryPublicKeysCount = uint64
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply schemas/state/multisig_entry.cats:15]: good catch i forgot to update that one :+1:

**Comment by Jaguar0625** [comment]
> Is it technically possible to change signerPublicKey fields to addresses? Will entities be less verifiable if this change is applied?

we want to be able to statelessly validate the signature. think of the signer + signature as a header that contains all inputs to crypto verify.

File: schemas/state/secret_lock.cats (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgarcia360) [reply schemas/state/secret_lock.cats:7]: ok, sounds good 👍

**Pull request closed without merging**